File: src/Entity/OffreStage.php
<?php

namespace App\Entity;

use App\Repository\OffreStageRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;


/**
 * @ORM\Entity(repositoryClass=OffreStageRepository::class)
 */
class OffreStage
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank
     */
    private $NomEntreprise;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank
     */
    private $Description;

    /**
     * @Assert\NotBlank
     * @ORM\Column(type="string", length=255)
     */
    private $stage_id;

    /**
     * @ORM\OneToMany(targetEntity=CandidatureStage::class, mappedBy="id_stage",cascade={"persist"})
     */
    private $relation;

    /**
     * @ORM\OneToOne(targetEntity=Quiz2::class, cascade={"persist"})
     */
    private $finder;
    /*
    /**
     * @ORM\OneToOne(targetEntity=Quiz2::class, mappedBy="Id_stageQ")
     */
    //private $relation2;

    public function __construct()
    {
        $this->relation = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomEntreprise(): ?string
    {
        return $this->NomEntreprise;
    }

    public function setNomEntreprise(string $NomEntreprise): self
    {
        $this->NomEntreprise = $NomEntreprise;

        return $this;
    }



    public function getDescription(): ?string
    {
        return $this->Description;
    }

    public function setDescription(string $Description): self
    {
        $this->Description = $Description;

        return $this;
    }

    public function getStageId(): ?string
    {
        return $this->stage_id;
    }

    public function setStageId(string $stage_id): self
    {
        $this->stage_id = $stage_id;

        return $this;
    }
    /**
     * @return Collection|CandidatureStage[]
     */
    public function getRelation(): Collection
    {
        return $this->relation;
    }

    public function addRelation(CandidatureStage $relation): self
    {
        if (!$this->relation->contains($relation)) {
            $this->relation[] = $relation;
            $relation->setIdStage($this);
        }

        return $this;
    }

    public function removeRelation(CandidatureStage $relation): self
    {
        if ($this->relation->removeElement($relation)) {
            // set the owning side to null (unless already changed)
            if ($relation->getIdStage() === $this) {
                $relation->setIdStage(null);
            }
        }

        return $this;
    }
/*
    /**
     * @return Collection|Quiz2[]
     */
   /*
    public function getRelation2(): Collection
    {
        return $this->relation2;
    }

    public function addRelation2(Quiz2 $relation2): self
    {
        if (!$this->relation2->contains($relation2)) {
            $this->relation2[] = $relation2;
            $relation2->setIdStageQ($this);
        }

        return $this;
    }

    public function removeRelation2(Quiz2 $relation2): self
    {
        if ($this->relation2->removeElement($relation2)) {
            // set the owning side to null (unless already changed)
            if ($relation2->getIdStageQ() === $this) {
                $relation2->setIdStageQ(null);
            }
        }

        return $this;
    }*/
    public function __toString()
    {
        return (string) $this->stage_id;
    }

    public function getFinder(): ?Quiz2
    {
        return $this->finder;
    }

    public function setFinder(?Quiz2 $finder): self
    {
        $this->finder = $finder;

        return $this;
    }



}

File: src/Stage/entities/OffreStage.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Stage.entities;

/**
 *
 * @author imote
 */
public class OffreStage {
    private int id;
    private String nom_entreprise;
    private String description;
    private String stage_id;

    public OffreStage() {
    }

    public OffreStage(int id, String nom_entreprise, String description, String stage_id) {
        this.id = id;
        this.nom_entreprise = nom_entreprise;
        this.description = description;
        this.stage_id = stage_id;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom_entreprise() {
        return nom_entreprise;
    }

    public void setNom_entreprise(String nom_entreprise) {
        this.nom_entreprise = nom_entreprise;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getStage_id() {
        return stage_id;
    }

    public void setStage_id(String stage_id) {
        this.stage_id = stage_id;
    }

    @Override
    public String toString() {
        return stage_id + " proposé par " + nom_entreprise;
    }
   
    
}
